framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -1252,6 +1262,115 @@ MeshDiagnosticsGenerator::checkLocalJacobians(const std::unique_ptr<MeshBase> &
                  num_negative_side_qp_jacobians);
 }
 
+void
+MeshDiagnosticsGenerator::checkNonMatchingEdges(const std::unique_ptr<MeshBase> & mesh) const
+{
+  /*Algorithm Overview
+    1)Prechecks
+      a)This algorithn only works for 3D so check for that first
+    2)Loop
+      a)Loop through every element
+      b)For each element get the edges associated with it
+      c)For each edge check overlap with any edges of nearby elements
+      d)Have check to make sure the same pair of edges are not being tested twice for overlap
+    3)Overlap check
+      a)Shortest line that connects both lines is perpendicular to both lines
+      b)A good overview of the math for finding intersecting lines can be found
+    here->paulbourke.net/geometry/pointlineplane/
+  */
+  if (mesh->mesh_dimension() != 3)
+    mooseError("The edge intersection algorithm only works with 3D meshes");
+  if (!mesh->is_serial())
+    mooseError("Only serialized/replicated meshes are supported");
+  unsigned int num_intersecting_edges = 0;
+  std::set<Node> checked_edges_nodes;
```

Comment:
```suggestion
  std::set<Node> overlapping_edges_nodes;
```
the "checked" part is what made me think we were storing every single node there